### ISSUE TITLE
Auto save option issue

### ISSUE THREAD

**Issue by viznov** [ReDEnergy/SessionSync#66]
Could anyone help with autosave option? I have no clue where auto saved sessions are stored. I see only manually saved sessions. Auto saving is on, the interval is 60 sec, max entries 60, deleting after 10 hours.
Thank you!

**Comment by ReDEnergy** [comment]
You should be able to see a History Tab (that blue one selected from the image below) . After you click on it you should be able to see a list of saved sessions. 
![image](https://user-images.githubusercontent.com/1395510/54319492-61663080-45f2-11e9-8539-0ce145596ca5.png)

Sadly I'm still working on the improved version of auto-save so what you currently will only save information over the same sessions again and again. Only after you close the browser completely and start it again a new auto-save session will be created. I know that it's not ideal and I have a working version that I'm testing right now that greatly improves the functionality. Sadly I still miss the proper UI so I still have some work on it before it's ready.

**Comment by viznov** [comment]
Thank you for your answer!
I just reinstalled addon but still have no saved items in the tab.
https://www.dropbox.com/s/med4pjsd8x1fdng/2019-03-13_19-45-56.png?dl=0
What problem it could be?

**Comment by ReDEnergy** [comment]
Again... sadly the current version will only replace the last session every 15 seconds (like you set) until you close the browser and then restart it. Basically it doesn't create a new session every 15 seconds but replaced the "active sessions" because that's what I considered at the time I implemented it (know I know it's not something everyone is expecting but like I said... I still have some things to do before I can launch the updated version).
For example if you do this steps you should be able to see more than session there:
 - (optional)  manually save the session just to make (you don't lose stuff)
 - make sour there is something saved in that "Current session"... it should as I can see in the image you sent
 - close the browser ()
 - open the browser and you should be able to see 2 session in the history session
   - the first one "Current session" will held information about the active session from when you started the browser
   - the second one should be the previous one saved at some point before you closed the browser

Sorry if it's not what how you were expecting it to work.
Basically, that's how it works for the moment until I launch the improved version that will actually save everything on a rolling buffer (each time will use a different saving slot to update the session).

**Comment by viznov** [comment]
I got it. Thank you for help and detailed answer!

**Comment by pekohub** [comment]
Can I ask, what's the progress in the Auto-Save feature? I see the same behaviour is still active, so is there a chance to get it anytime soon?

--
Cheers,
Peter.

**Comment by ReDEnergy** [comment]
@pekohub So the auto-saving right now saves every X seconds (configurable) over the same session slot. Until the browser is not completely closed and reopened the slot is not changed.
Of course since the saving is done again and again over the same session you can lost things from the past very easy.

I let this issue opened because I'm suppose to offer an option to select how many entries should be used for saving and not replace everything always.

From what I remember I actually had something like this a while ago but I had some design problems with it. Like, how many session should be saved. I can't possible save everything every 30 seconds because then if you leave the browser opened for 10 hours you would probably end with 1200 sessions and I'm pretty sure even displaying so many sessions would be a problem. So I have to add an option for the user to select the maximum number of active saving sessions he wants to keep, so that I can reuse the old one.

Basically, if you set to save every 30 seconds and chose only 10 slots then all your history will be preserved each 30 seconds for a total of 300 seconds which means 5 minutes. Everything older will be lost. Of course you will be able to save everything each 5 minutes with 20 slots if you would like to... so that would be enough for 2 hours worth of browsing. Or how much do you want.

When will this be ready... I actually can't tell you exactly because I know I had an issue when working on it but probably after than with various other stuff going on (jobs, several other projects) I kind of forgot about it.

I do have to launch an update soon for session replacing so I will look more into this to see if there's anything I can do to launch it at the same time.

**Comment by pekohub** [comment]
Thanks for the update, @ReDEnergy. 
Personally, what I'm looking for in this feature is a way to restore what tabs I had at a certain time, where the overall period could be a few days, and within these days, once an hour would be totally enough for me. The reason to do this? I'm also a tab hoarder, and tabs kind of remind me of things I wanted to look at or even do something. When a browser crash occurs, sometimes the session restores correctly, but sometimes it does not, and that's where the tab history would kick in ;)

**Comment by ReDEnergy** [comment]
Yes, It should be able to do that as long as you configure it the way you want: save every 1 hour, delete after 5 days or after a certain number of session is reached. Hope I can push an update with this soon enough.
